server/paymentProviders/stripe/virtual-cards.js

```diff
@@ -86,9 +86,21 @@ export const processAuthorization = async (stripeAuthorization, stripeEvent) =>
 
   await checkStripeEvent(host, stripeEvent);
 
-  // TODO : convert balance to the same currency as amount
-  const amount = convertToStripeAmount(host.currency, stripeAuthorization.amount);
-  const balance = await host.getBalanceWithBlockedFundsAmount();
+  const existingExpense = await models.Expense.findOne({
+    where: {
+      VirtualCardId: virtualCard.id,
+      data: { authorizationId: stripeAuthorization.id },
+    },
+  });
+
+  if (existingExpense) {
+    logger.warn(`Virtual Card authorization already reconciled, ignoring it: ${stripeAuthorization.id}`);
+    return;
+  }
+
+  const currency = stripeAuthorization.currency.toUpperCase();
```

Comment:
Do we have a guarantee that this will always be the currency of the card? If you have something, could you add a link to the documentation of what to expect.

Reply:
In the end, doesn't matter it seems, as we're handling everything properly after. Could be any currency and the code would handle this right.